src/Nethermind/Nethermind.Serialization.Rlp/BlockInfoDecoder.cs

```diff
@@ -51,35 +51,56 @@ public class BlockInfoDecoder : IRlpStreamDecoder<BlockInfo>, IRlpValueDecoder<B
             return blockInfo;
         }
 
-        public void Encode(RlpStream stream, BlockInfo item, RlpBehaviors rlpBehaviors = RlpBehaviors.None)
+        public void Encode(RlpStream stream, BlockInfo? item, RlpBehaviors rlpBehaviors = RlpBehaviors.None)
         {
-            throw new NotImplementedException();
+            if (item == null)
+            {
+                stream.Encode(Rlp.OfEmptySequence);
+                return;
+            }
+
+            int contentLength = GetContentLength(item, rlpBehaviors);
+
+            bool hasMetadata = item.Metadata != BlockMetadata.None;
+            stream.StartSequence(contentLength);
+            stream.Encode(item.BlockHash);
+            stream.Encode(item.WasProcessed);
+            stream.Encode(item.TotalDifficulty);
+            if (hasMetadata)
+            {
+                stream.Encode((int)item.Metadata);
+            }
         }
 
         public Rlp Encode(BlockInfo? item, RlpBehaviors rlpBehaviors = RlpBehaviors.None)
         {
-            if (item is null)
+            throw new NotImplementedException();
```

Comment:
Can we just remove it?

Reply:
this is due to the fact the BlockInfoDecoder implement IRlpValueDecoder

Reply:
lets make: `interface IRlpValueDecoder<T> : IRlpDecoder<T>` ?